src/Compilers/CSharp/Portable/BoundTree/BoundDagTest.cs

```diff
@@ -43,7 +43,7 @@ private bool Equals(BoundDagTest? other)
 
         public override int GetHashCode()
         {
-            return Hash.Combine(Kind.GetHashCode(), Input.GetHashCode());
+            return Hash.Combine(((int)Kind).GetHashCode(), Input.GetHashCode());
```

Comment:
Is there something we can do to avoid this in general?

Reply:
We could maybe use Banned API Analyzer to ban `Enum.GetHashCode`

Reply:
> Is there something we can do to avoid this in general?

I assume the execution where this was observed to be doing lots of boxing was on Framework? I'd expect no boxing to be happening here with the simpler code on Core with optimized code.

Reply:
Correct, this is .NET Framework. I also mentioned this in the banned API file.

Reply:
Yes this was addressed in .NET Core: https://github.com/dotnet/coreclr/pull/7895.